src/GameLogic/PlugIns/ChatCommands/ChatCommandPlugInBase.cs

```diff
@@ -174,6 +174,25 @@ protected ValueTask ChangeAccountStateByLoginNameAsync(Player gameMaster, string
         return this.ChangeAccountStateAsync(gameMaster, (MUnique.OpenMU.Persistence.IPlayerContext context) => context.GetAccountByLoginNameAsync(loginName), accountState);
     }
 
+    /// <summary>
+    /// Changes ChatBanUntil value from Account.
+    /// </summary>
+    /// <param name="gameMaster">GameMaster Player.</param>
+    /// <param name="account">Account to be changed.</param>
+    protected async ValueTask ChangeAccountChatBanUntilAsync(Player gameMaster, Account? account, DateTime? chatBanUntil)
+    {
+        if (account != null)
+        {
+            account.ChatBanUntil = chatBanUntil;
+            using var context = gameMaster.GameContext.PersistenceContextProvider.CreateNewPlayerContext(gameMaster.GameContext.Configuration);
```

Comment:
@sven-n 

If I understand the functioning of periodic saving correctly, there is a potential issue with removing these lines.
In the event that the server becomes unresponsive (such as in the case of a crash or power loss), any bans or unbans that were performed in the time window between periodic saves will not be saved.

Let's consider an example:

- The server has the PeriodicSave feature configured with an interval of 1 minute.
- A GM executes the /chatban command specifying a duration of 60 minutes.
- Before the next periodic save occurs, the server is manually stopped.
- The server is restarted
- The player who should have been chat banned is not actually banned.

Should i remove these lines?
Thanks!

Reply:
Well, that's a point. But the problem with crashes/power loss is, that also other player progress is not saved in this case. Crashes should be prevented under any circumstance.
About manually stopping and restarting: When the server is stopped gracefully in the admin panel in the server list first, it would save all player contexts.
Another problem with manually triggering the save: The periodic save considers the state of the player, so it saves only when the player is in the "EnteredWorld" state. E.g. during a trade, the save should not occur, because the assigned item slots are not inventory slots, but trade box slots.
So, I would like to keep that simple, not saving it explicitly.

Reply:
I understand, thanks for the detailed explanation. I removed the save in https://github.com/MUnique/OpenMU/pull/360/commits/640be73960baf88fecd58df6641cf21faa391817